### ISSUE TITLE
Tray icon isn't working

### ISSUE THREAD

**Issue by lazka** [flathub/org.zulip.Zulip#10]
Not sure if it is supposed to work or not, but enabling it in the settings doesn't do anything in an Ubuntu GNOME Shell environment.

Is it working for anyone else and is it working with the official distribution?

**Comment by lazka** [comment]
ok, missing `--talk-name=org.kde.StatusNotifierWatcher`, but the icon is invisible because it exports the wrong name i guess?

```python
{'AttentionAccessibleDesc': '',
 'AttentionIconName': '',
 'Category': 'ApplicationStatus',
 'IconAccessibleDesc': 'icon',
 'IconName': 'Zulip1_10',
 'IconThemePath': '/tmp/.org.chromium.Chromium.ByGE72',
 'Id': 'Zulip1',
 'Menu': '/org/ayatana/NotificationItem/Zulip1/Menu',
 'Status': 'Active',
 'Title': 'zulip',
 'XAyatanaLabel': '',
 'XAyatanaLabelGuide': '',
 'XAyatanaOrderingIndex': 0}
```

**Comment by ghisvail** [comment]
I would suggest to close this issue since the tray icon is now working. The missing icon is a separate issue.

**Comment by devurandom** [comment]
Where is the problem with the missing icon being tracked?

**Comment by lazka** [comment]
Nowhere atm I think. #2 might be a way forward since we then could patch things easily.

**Comment by ghisvail** [comment]
The flatpak for Zulip is now built from source. Feel free to experiment with patches to fix the tray icon issue.

Future PR will be warmly welcome :-)

**Comment by lazka** [comment]
uh, very nice :D

**Comment by lazka** [comment]
and looks like my 5.0.0 update broke tray icons altogether again. I'll look into it some time, but no promises :)

**Comment by lazka** [comment]
hm, somehoe it crashes now when launching from gnome shell, "flatpak run" works:

```
Apr 04 16:59:38 xy systemd[1762]: Started Application launched by gnome-shell.
Apr 04 16:59:38 xy systemd[1762]: Started flatpak-org.zulip.Zulip-97646.scope.
Apr 04 16:59:38 xy systemd[1762]: gnome-launched-org.zulip.Zulip.desktop-97646.scope: Succeeded.
Apr 04 16:59:38 xy org.zulip.Zulip.desktop[97661]: [2:0404/165938.570713:FATAL:setuid_sandbox_host.cc(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /app/bin>
Apr 04 16:59:38 xy kernel: traps: zulip[97661] trap int3 ip:556398b80953 sp:7ffd81dd96c0 error:0 in zulip[556396d0a000+53cf000]
Apr 04 16:59:38 xy systemd[1]: Started Process Core Dump (PID 97663/UID 0).
Apr 04 16:59:38 xy systemd[1762]: flatpak-org.zulip.Zulip-97646.scope: Succeeded.
Apr 04 16:59:38 xy systemd-coredump[97664]: Process 97661 (zulip) of user 1000 dumped core.
                                            
                                            Stack trace of thread 2:
                                            #0  0x0000556398b80953 n/a (/app/bin/zulip + 0x354b953)
                                            #1  0x000055639a0f6ca8 n/a (/app/bin/zulip + 0x4ac1ca8)
                                            #2  0x000055639a0f612d n/a (/app/bin/zulip + 0x4ac112d)
                                            #3  0x00005563983ca7b8 n/a (/app/bin/zulip + 0x2d957b8)
                                            #4  0x000055639a0f54be n/a (/app/bin/zulip + 0x4ac04be)
                                            #5  0x000055639a0fb241 n/a (/app/bin/zulip + 0x4ac6241)
                                            #6  0x00005563983ca0d3 n/a (/app/bin/zulip + 0x2d950d3)
                                            #7  0x000055639a0fbc8c n/a (/app/bin/zulip + 0x4ac6c8c)
                                            #8  0x000055639777dff1 n/a (/app/bin/zulip + 0x2148ff1)
                                            #9  0x0000556396d0a8fb n/a (/app/bin/zulip + 0x16d58fb)
                                            #10 0x00007fc6983c5173 n/a (/usr/lib/x86_64-linux-gnu/libc-2.30.so + 0x24173)
Apr 04 16:59:38 xy systemd[1]: systemd-coredump@7-97663-0.service: Succeeded.
```

**Comment by lazka** [comment]
The desktop file still exports the wrong command somehow

`cat /var/lib/flatpak/exports/share/applications/org.zulip.Zulip.desktop`

```
[Desktop Entry]
Name=Zulip
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=zulip --file-forwarding org.zulip.Zulip @@u %U @@
Terminal=false
Type=Application
Icon=org.zulip.Zulip
StartupWMClass=Zulip
Comment=Zulip Desktop Client for Linux
Categories=Chat;GNOME;GTK;Network;InstantMessaging;
X-Desktop-File-Install-Version=0.24
X-Flatpak-RenamedFrom=zulip.desktop;
X-Flatpak=org.zulip.Zulip
```

**Comment by devurandom** [comment]
The currently published version is 5.0.0 from the official binary, not the one built from source: https://flathub.org/builds/#/apps/org.zulip.Zulip

For me, `flatpak run org.zulip.Zulip` does not start the app (or starts it, but there is no tray icon at all and no window popping up), so I send SIGINT (^C) after a while:
```
❯ flatpak info org.zulip.Zulip

Zulip - Zulip Desktop Client for Linux

          ID: org.zulip.Zulip
         Ref: app/org.zulip.Zulip/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 5.0.0
     License: Apache-2.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 218.5 MB
     Runtime: org.freedesktop.Platform/x86_64/19.08
         Sdk: org.freedesktop.Sdk/x86_64/19.08

      Commit: 0f5ffde5118285984cf4dd983200570efcb0062af688dde8d44de6608480a551
      Parent: 2d18d6b119bc26ec87a24fabbd08765e3560353742efba64cdca95211c79d6dd
     Subject: Update to 5.0.0 (532c515d)
        Date: 2020-04-03 19:12:31 +0000

❯ flatpak run org.zulip.Zulip
(electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
(electron) 'setUserAgent function' is deprecated and will be removed. Please use 'userAgent property' instead.
(electron) 'setAudioMuted function' is deprecated and will be removed. Please use 'audioMuted property' instead.
^C⏎
[2:0404/175247.547582:FATAL:bus.cc(1220)] D-Bus connection was disconnected. Aborting.
Failed to generate minidump.~ took 1m57s
```

Running it from KRunner (KDE Plasma) makes the process just run for a few seconds and then exit (checked with `flatpak ps` on the terminal).

I cannot update to the latest unpublished build either:
```
❯ sudo flatpak install --or-update https://dl.flathub.org/build-repo/17181/org.zulip.Zulip.flatpakref
error: App org.zulip.Zulip, branch stable is already installed
❯ sudo flatpak install --reinstall https://dl.flathub.org/build-repo/17181/org.zulip.Zulip.flatpakref
error: App org.zulip.Zulip, branch stable is already installed
```

I will wait 24h for the new build to be published (https://blogs.gnome.org/alexl/2019/02/19/changes-in-flathub-land/, "New publish workflow") and then try again.

**Comment by ghisvail** [comment]
Yes, the build from source will use the zyppak-wrapper which solves the issue with sandboxing in Electron.

**Comment by lazka** [comment]
I've filed https://github.com/flatpak/flatpak/issues/3534

**Comment by devurandom** [comment]
I am now on the built-from-source version, but still the same behaviour (no tray icon at all and no window popping up) as in https://github.com/flathub/org.zulip.Zulip/issues/10#issuecomment-609049069:
```
❯ flatpak info org.zulip.Zulip

Zulip - Zulip Desktop Client for Linux

          ID: org.zulip.Zulip
         Ref: app/org.zulip.Zulip/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 5.0.0
     License: Apache-2.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 454.3 MB
     Runtime: org.freedesktop.Platform/x86_64/19.08
         Sdk: org.freedesktop.Sdk/x86_64/19.08

      Commit: 3d6f107f27ed5227505dbca7018187fd6e6b1bbb9c819b39562d76371f165ca7
      Parent: c29f683a52ee5e7308da7c7d3e213bf8e4f9d021475b1a38aa1604bc1b3f9c2a
     Subject: Enable builds for 64-bit ARM (efc63174)
        Date: 2020-04-04 15:06:29 +0000

❯ flatpak run org.zulip.Zulip
(electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
(electron) 'setUserAgent function' is deprecated and will be removed. Please use 'userAgent property' instead.
(electron) 'setAudioMuted function' is deprecated and will be removed. Please use 'audioMuted property' instead.
^C[2:0405/081420.155716:FATAL:bus.cc(1220)] D-Bus connection was disconnected. Aborting.
Failed to generate minidump.~ took 2m12s 
```

**Comment by ghisvail** [comment]
My bad I forgot to include a desktop file, since the source build no longer provides one. That's why you no longer see Zulip listed in your application launcher. Should be fixed with #17.
